python-avd/pyavd/_eos_designs/structured_config/core_interfaces_and_l3_edge/utils.py

```diff
@@ -295,11 +295,19 @@ def _get_ethernet_cfg(self: AvdStructuredConfigCoreInterfacesAndL3Edge, p2p_link
 
         ptp_config = {}
 
+        if (ptp_profile_name := get(p2p_link, "ptp.profile", default=self.shared_utils.ptp_profile_name)) is not None:
+            msg = f"PTP Profile '{ptp_profile_name}' referenced under {context}.p2p_links does not exist in `ptp_profiles`."
+            ptp_config.update(get_item(self.shared_utils.ptp_profiles, "profile", ptp_profile_name, required=True, custom_error_msg=msg))
+
         # Apply PTP profile config
         if self.shared_utils.ptp_enabled:
             ptp_config.update(self.shared_utils.ptp_profile)
 
         ptp_config["enable"] = True
+
+        if get(p2p_link, "role") != "bmca":
```

Comment:
it means default role is master which is not correct

Reply:
I saw similar logic in connected_endpoints.utils.

Reply:
We should have different assumptions for endpoints and p2p links.

Reply:
This logic behaves different than the schema says. The schema says the default is "bmca", but your logic requires "bmca" to be set directly.

Reply:
Changed bmca to dynamic and corrected the default output.